enum ranges

```diff
@@ -0,0 +1,50 @@
+import Data.Time.Calendar
+
+
+data Rank = R2 | R3 | R4 | R5 | R6 | R7 | R8 | R9 | R10 | J | Q | K | A
+            deriving (Bounded, Enum, Eq, Ord)
```

Comment:
We noticed this comes from your haskell-exercises repo, it's cool that you had an example handy to use :)

Instead of single-letter names for the contructors, maybe entire words would be more clear? It took us a minute to figure out what J, Q, K, A meant; I think seeing "jack", "queen", etc. could make it more immediately obvious that we're talking about playing cards.

```haskell
data Rank = Rank2 | Rank3 | Rank4 | Rank5 | Rank6 | Rank7
          | Rank8 | Rank9 | Rank10 | Jack | Queen | King | Ace
```